pkg/errhelp/errors.go

```diff
@@ -148,6 +149,9 @@ func NewAzureError(err error) error {
 }
 
 func NewAzureErrorAzureError(err error) *AzureError {
+	if err == nil {
```

Comment:
Nice. The name of this totally threw me but I can see why it's like that.

Reply:
The name _completely_ mystifies me - and I've spent 15m looking at this and the use of it to try and get my head around it.

At minimum, it needs a comment to explain what it does; better if it's given a name that actually expresses what it does.

Reply:
Given other Go code that accepts `error` and returns something else, maybe the name needs to start with `Wrap`?

Reply:
I've renamed this to `NewAzureError` and combined this and the existing `NewAzureError` since there's no reason to have the method that returns error (since `*AzureError` impls it anyway).